[BUG-7732][fix] fix column 'is_directory' of table t_ds_resources typ…

jim-parsons opened the pull request:
<!--Thanks very much for contributing to Apache DolphinScheduler. Please review https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html before opening a pull request.-->


## Purpose of the pull request

[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

1. add `Boolean2IntTypeHandler.java` both in class `Resource.java` and `ResourceMapper.xml`
2. specify typeHandler packages and update session config in class `SpringConnectionFactory` and remove the deprecated bean `PaginationInterceptor`
3. format class `Resource`
4. update `mybatis-plus` version from `3.2.0` to `3.4.0`. The current version won’t pass maven test of this [MYBATIS-BUG](https://github.com/baomidou/mybatis-plus/issues/2550). 
5. due to 'isDirector' is primitive type, so the test won't cover the wrapper type's null situation.
6. update `known-dependencies.txt`

This closes #7732

<!--(For example: This pull request adds checkstyle plugin).-->

## Brief change log

<!--*(for example:)*
  - *Add maven-checkstyle-plugin to root pom.xml*
-->
## Verify this pull request

<!--*(Please pick either of the following options)*-->

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

<!--*(example:)*
  - *Added dolphinscheduler-dao tests for end-to-end.*
  - *Added CronUtilsTest to verify the change.*
  - *Manually verified the change by testing locally.* -->

jim-parsons committed:
[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

1. add Boolean2IntTypeHandler both in class Resource.java and ResourceMapper.xml
2. specify typeHandler packages in class SpringConnectionFactory
3. format class Resource
4. update `mybatis-plus` version from `3.2.0` to `3.4.0`. The current version won’t pass maven test of this [MYBATIS-BUG](https://github.com/baomidou/mybatis-plus/issues/2550). Besides update the config of `SpringConnectionFactory`
5. due to 'isDirector' is primitive type, so the test won't cover the wrapper type' null situation.

This closes #7732

jim-parsons committed:
[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

1. add Boolean2IntTypeHandler both in class Resource.java and ResourceMapper.xml
2. specify typeHandler packages in class SpringConnectionFactory
3. format class Resource
4. update `mybatis-plus` version from `3.2.0` to `3.4.0`. The current version won’t pass maven test of this [MYBATIS-BUG](https://github.com/baomidou/mybatis-plus/issues/2550). Besides update the config of `SpringConnectionFactory`
5. due to 'isDirector' is primitive type, so the test won't cover the wrapper type' null situation.

This closes #7732

jim-parsons committed:
[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

1. add Boolean2IntTypeHandler both in class Resource.java and ResourceMapper.xml
2. specify typeHandler packages in class SpringConnectionFactory
3. format class Resource
4. update `mybatis-plus` version from `3.2.0` to `3.4.0`. The current version won’t pass maven test of this [MYBATIS-BUG](https://github.com/baomidou/mybatis-plus/issues/2550). Besides update the config of `SpringConnectionFactory`
5. due to 'isDirector' is primitive type, so the test won't cover the wrapper type' null situation.

This closes #7732

jim-parsons committed:
[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

1. add Boolean2IntTypeHandler both in class Resource.java and ResourceMapper.xml
2. specify typeHandler packages in class SpringConnectionFactory
3. format class Resource
4. update `mybatis-plus` version from `3.2.0` to `3.4.0`. The current version won’t pass maven test of this [MYBATIS-BUG](https://github.com/baomidou/mybatis-plus/issues/2550). Besides update the config of `SpringConnectionFactory`
5. due to 'isDirector' is primitive type, so the test won't cover the wrapper type' null situation.

This closes #7732

ruanwenjun commented on pom.xml:
```diff
@@ -60,7 +60,7 @@
         <quartz.version>2.3.2</quartz.version>
         <jackson.version>2.10.5</jackson.version>
-        <mybatis-plus.version>3.2.0</mybatis-plus.version>
+        <mybatis-plus.version>3.4.0</mybatis-plus.version>
```
It's better not upgrade `mybatis-plus` in this PR, If you hope to do this, it's necessary to create a new issue and discuss with community first.

jim-parsons replied:
sorry about that, i choose the way of adding a typeHandler to fix issue #7732 . but due to the mybatis-plus [BUG](https://github.com/baomidou/mybatis-plus/issues/2550), it failed in the maven test. so i update a oldest version that works.
  
  but if there is anthor way to fix this bug, i think i won't change the version

ruanwenjun replied:
We can discuss in #7732 .

jim-parsons committed:
[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

1. add Boolean2IntTypeHandler both in class Resource.java and ResourceMapper.xml
2. specify typeHandler packages in class SpringConnectionFactory
3. format class Resource
4. update `mybatis-plus` version from `3.2.0` to `3.4.0`. The current version won’t pass maven test of this [MYBATIS-BUG](https://github.com/baomidou/mybatis-plus/issues/2550). Besides update the config of `SpringConnectionFactory`
5. due to 'isDirector' is primitive type, so the test won't cover the wrapper type' null situation.
6. update `known-dependencies.txt`

This closes #7732

jim-parsons committed:
[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

1. add Boolean2IntTypeHandler both in class Resource.java and ResourceMapper.xml
2. specify typeHandler packages in class SpringConnectionFactory
3. format class Resource
4. update `mybatis-plus` version from `3.2.0` to `3.4.0`. The current version won’t pass maven test of this [MYBATIS-BUG](https://github.com/baomidou/mybatis-plus/issues/2550). Besides update the config of `SpringConnectionFactory`
5. due to 'isDirector' is primitive type, so the test won't cover the wrapper type' null situation.
6. update `known-dependencies.txt`

This closes #7732

jim-parsons committed:
[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

[FIX-7732][fix] fix column 'is_directory' of table t_ds_resources type error in PG database

1. add `Boolean2IntTypeHandler.java` both in class `Resource.java` and `ResourceMapper.xml`
2. specify typeHandler packages and update session config in class `SpringConnectionFactory` and remove the deprecated bean `PaginationInterceptor`
3. format class `Resource`
4. update `mybatis-plus` version from `3.2.0` to `3.4.0`. The current version won’t pass maven test of this [MYBATIS-BUG](https://github.com/baomidou/mybatis-plus/issues/2550).
5. due to 'isDirector' is primitive type, so the test won't cover the wrapper type's null situation.
6. update `known-dependencies.txt`

This closes #7732

jim-parsons commented:
this pr can close, thx, i've create a new one.
